### ISSUE TITLE
Code refactoring

### ISSUE THREAD

**Issue by romainx** [JosPolfliet/pandas-profiling#65]
Hello,

I would like to perform a code refactoring in order to split the big `base.py` module in several modules with separated concerns:
- `plot`: Dedicated to plot features (histograms at this time)
- `describe`: Dedicated to compute the variables description
- `report`: Dedicated to generate the report
- `base`: Containing only remaining common features

Additional modules will be ket _as is_.
- `formatters`: Containing formatting utilities
- `templates`: Jinja templating helpers and definition

It's just a first step, then I would like to split the big report generation, but this is another story ;-)

I would like your approval before starting it, since it will change a lot of things.

Many thanks for your answer.
Best,
Romain

**Comment by JosPolfliet** [comment]
Sounds like a good idea. It has grown a bit out of the reasonable module length, for sure. If you submit I pull request I will review and check if all works as supposed to. The automated `tox` tests can help too, but they only find functional errors, no aesthetical errors.

**Comment by romainx** [comment]
Great! I will work on it carefully and let you know.

**Comment by romainx** [comment]
I close this issue since the main work has been done. I will open other PR in order to perform other refactoring tasks.
